packages/terra-paginator/src/ProgressivePaginator.jsx

```diff
@@ -77,17 +81,38 @@ class ProgressivePaginator extends React.Component {
       intl,
       totalCount,
       itemCountPerPage,
+      pageLabel,
     } = this.props;
     const totalPages = (totalCount) ? calculatePages(totalCount, itemCountPerPage) : 0;
     const { selectedPage } = this.state;
     const previousPageIndex = selectedPage === 1 ? 1 : selectedPage - 1;
     const nextPageIndex = selectedPage === totalPages ? totalPages : selectedPage + 1;
 
+    let messageId;
+    let messageAttributes;
+
+    if (totalCount) {
+      if (pageLabel) {
+        messageId = 'Terra.paginator.pageCountWithLabel';
+        messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages };
+      } else {
+        messageId = 'Terra.paginator.pageCount';
+        messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages };
+      }
+    } else if (pageLabel) {
+      messageId = 'Terra.paginator.pageIndexWithLabel';
+      messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages };
+    } else {
+      messageId = 'Terra.paginator.pageIndex';
+      messageAttributes = { pageLabel, pageNumber: selectedPage, pageNumberTotal: totalPages };
+    }
+
```

Comment:
Can we create helper function and accept this params to return messageAttributes for all the variants..???

Reply:
Added here https://github.com/cerner/terra-core/pull/3470/commits/3cd856ed21195a75b96626b494560a4505bd628a